### PR TITLE
Refactor/event types

### PR DESCRIPTION
**Pull request by mrkarimoff**
This PR intends to make the following changes: 
- Adds optional `metadata` field to `AnalyticsEventBase` to allow both `AnalyticsEvent` and `AnalyticsError` to have metadata properties
 - Entends  `AnalyticsError` from  `AnalyticsEventBase` to have an optional metadata field and makes the `error` property the JavaScript `Error`
 - Inserts analytics error metadata into the Database
 - Adds Drizzle ORM integration for DB communication to the analytics website
 - Adds Filter for the event table on the analytics dashboard
 - Removes tabs and adds one more info box for `DataExport` on the dashboard

### CODE REVIEW

File: apps/analytics/app/_components/analytics/EventsTable/EventsTable.tsx
Comment (jthrilly) [inline apps/analytics/app/_components/analytics/EventsTable/EventsTable.tsx:1]: It would have been preferable to leave this as a server page, and to use the search parameters to add filtering to the database query.

File: apps/analytics/app/api/event/route.ts [hunk 1]
```diff
@@ -13,36 +14,27 @@ export const runtime = "edge";
...
   const event = (await request.json()) as DispatchableAnalyticsEvent;
 
-  const timestamp = JSON.stringify(event.timestamp || new Date().toISOString());
+  let generalEvent: EventInsertType = {
+    type: event.type,
+    metadata: event.metadata,
+    timestamp: event.timestamp,
+    installationId: event.installationId,
+    isocode: event.geolocation?.countryCode,
+  };
```
Comment (jthrilly) [inline apps/analytics/app/api/event/route.ts:23]: This should use the types from the analytics package! If these are needed for validation, consider updating the analytics types to be derived from a zod schema.

File: apps/analytics/app/api/event/route.ts [hunk 2]
```diff
@@ -13,36 +14,27 @@ export const runtime = "edge";
...
-        { error },
-        { status: 500, headers: corsHeaders }
-      );
-    }
+    generalEvent = {
+      ...generalEvent,
+      message: event.error.message,
+      name: event.error.name,
+      stack: event.error.stack,
+    };
```
Comment (jthrilly) [inline apps/analytics/app/api/event/route.ts:31]: You are only inserting events if they are type Error!!

What about normal analytics events?!

The whole point of this refactor was to eliminate different codepaths for errors and other analytics events!!

File: apps/analytics/scripts/seed.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+const eventTypes = [
+  "AppSetup",
+  "ProtocolInstalled",
+  "InterviewStarted",
+  "InterviewCompleted",
+  "InterviewCompleted",
+  "DataExported",
+  "Error",
+];
```
Comment (jthrilly) [inline apps/analytics/scripts/seed.ts:22]: Once again, this should be reused from a SINGLE source of truth!!

File: apps/analytics/tsconfig.json [hunk 4]
```diff
@@ -1,12 +1,8 @@
 {
   "extends": "tsconfig/nextjs.json",
   "compilerOptions": {
-    "target": "es5",
-    "lib": [
-      "dom",
-      "dom.iterable",
-      "esnext"
-    ],
+    "target": "ES6",
```
Comment (jthrilly) [inline apps/analytics/tsconfig.json:4]: what is the reason for this change?

**Review by jthrilly — CHANGES_REQUESTED** [review]
There are some serious issues with this PR that need to be addressed before merging. Please discuss with me on Slack.

File: apps/analytics/tsconfig.json (hunk 4 above)
Comment (mrkarimoff) [reply apps/analytics/tsconfig.json:4]: Sorry for not putting the docs ref: https://github.com/drizzle-team/drizzle-orm/issues/803 it's an issue from Drizzle

File: apps/analytics/scripts/seed.ts (hunk 3 above)
Comment (mrkarimoff) [reply apps/analytics/scripts/seed.ts:22]: I didn't come up with this code sir it was copied from the old seed.js file, and here we are storing the values types in an array to be able to pass it to here `const type = faker.helpers.arrayElement(eventTypes);`, I know that `@codaco/analytics` package exports `EventTypes` as a type not a value, so maybe we should add another export that stores them in an array maybe?

Comment (mrkarimoff) [reply apps/analytics/scripts/seed.ts:22]: "In TypeScript, you can't directly convert a type to a value because types exist at compile time and are erased at runtime, while values exist at runtime"

File: apps/analytics/app/_components/analytics/EventsTable/EventsTable.tsx
Comment (mrkarimoff) [reply apps/analytics/app/_components/analytics/EventsTable/EventsTable.tsx:1]: I added the filter to the table columns which is a client-side component, but I could try to implement what you're asking if it's important, maybe by using server action? Please let me know if it's crucial.

File: apps/analytics/app/api/event/route.ts (hunk 1 above)
Comment (mrkarimoff) [reply apps/analytics/app/api/event/route.ts]: Here, `EventInsertType` is the type of data that we need to insert into the database, I can't use types from the analytics package here because the database schema doesn't match the analytics event type as it can be either Error or Event.
Please talk to me on Slack in case this doesn't make sense because I might have not properly expressed my thought because of my English

File: apps/analytics/app/api/event/route.ts (hunk 2 above)
Comment (mrkarimoff) [reply apps/analytics/app/api/event/route.ts:31]: Please update or remove this review sir, this almost gave me a heart attack. 😄

File: apps/analytics/scripts/seed.ts (hunk 3 above)
Comment (jthrilly) [reply apps/analytics/scripts/seed.ts]: I didn't ask you to put this into a type - you derive the type from the variable in this case. What I was getting at is that these should only be defined once, and we have a definition of them in the analytics package already.

The point about this coming from existing code is irrelevant - you are doing the refactor here.

File: apps/analytics/app/api/event/route.ts (hunk 2 above)
Comment (jthrilly) [reply apps/analytics/app/api/event/route.ts]: Haha! Okay, apologies. This turned out to be as a result of poor code style and not actually a bug.

**Review by jthrilly — APPROVED** [review]
